### PR TITLE
Added more support for tooltips

### PR DESCRIPTION
**Pull request by crnormand**
#1 (in a series).   This is the base code for the tooltip support.   It is a non-destructive addition of functionality that is required to support the tooltip logic in 'gcs' (which will be coming in a future pull request).

Rows can now indicate if they want to "always show a tooltip" (regardless of size limitations), which defaults to false.

TextCells will check this method to determine if they should show a tooltip, and if so, they will request the "tooltip" information from the row (instead of the generic "getData").   This allows subclasses to define what represents a tooltip to them.

This code can be included as is, and it will not change any current functionality (since the default behavior for returning a "toolitp" is to just to "getData").

I re-cloned from your repository, and then force pushed to my github, so there should not be a lot of extraneous commits.

### CODE REVIEW

File: src/com/trollworks/toolkit/ui/widget/outline/TextCell.java [hunk 1]
```diff
@@ -275,4 +275,20 @@ public boolean participatesInDynamicRowLayout() {
...
+     * @param row    The row.
+     * @param column The column.
+     * @param nullOK <code>true</code> if <code>null</code> may be returned.
+     * @return The tooltip of this cell as a string.
+     */
+    @SuppressWarnings("static-method")
+    protected String getToolTip(Row row, Column column, boolean nullOK) {
+        if (row != null) {
+            String text = row.getToolTip(column);
+            return text == null ? nullOK ? null : "" : text; //$NON-NLS-1$
```
Comment (richardwilkes) [inline src/com/trollworks/toolkit/ui/widget/outline/TextCell.java]: Please change this line to this, for better readability (I find multiple tertiary operators without parens hard to follow):
```
return (text == null && !nullOK) ? "" : text; //$NON-NLS-1$
```

Comment (crnormand) [reply src/com/trollworks/toolkit/ui/widget/outline/TextCell.java]: The code formatter is automatically removing the extra parens.   BTW, this is the same code that is in getData(), I just copied it from there.

If you want, I could break it up into a formal if/then statement like this:

            if (text == null) {
                return nullOK ? null : ""; //$NON-NLS-1$
            }
            return text;

Comment (richardwilkes) [reply src/com/trollworks/toolkit/ui/widget/outline/TextCell.java]: Either way is fine... just don't like the `xxx ? xxx ? xxx : xxx : xxx` style... even if it is currently present in some places throughout the source.

Comment (crnormand) [reply src/com/trollworks/toolkit/ui/widget/outline/TextCell.java]: The formatter still stripped the parens, so I went with the full blown if/then statement.
